src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -736,7 +735,7 @@ public bool IsReady(SocketAsyncContext context, out int observedSequenceNumber)
                 using (Lock())
                 {
                     observedSequenceNumber = _sequenceNumber;
-                    bool isReady = (_state == QueueState.Ready);
+                    bool isReady = (_state == QueueState.Ready) || (_state == QueueState.Stopped);
```

Comment:
Why did this change?

Reply:
I got a test failure because this didn't cause an operation to be tried synchronously when the queue was stopped. It failed somewhere later, I believe in a sync-over-async path.
By returning true for stopped, this will throw ODE when the operation is performed synchronously.

Reply:
I got the same test failure with https://github.com/dotnet/corefx/pull/39079: 
```
Process terminated. Assertion failed.
   at System.Net.Sockets.SocketAsyncContext.Register() in /_/src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs:line 1150
   at System.Net.Sockets.SocketAsyncContext.OperationQueue`1.StartAsyncOperation(SocketAsyncContext context, TOperation operation, Int32 observedSequenceNumber, CancellationToken cancellationToken) in /_/src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs:line 754
   at System.Net.Sockets.SocketAsyncContext.PerformSyncOperation[TOperation](OperationQueue`1& queue, TOperation operation, Int32 timeout, Int32 observedSequenceNumber) in /_/src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs:line 1225
   at System.Net.Sockets.SocketAsyncContext.SendFile(SafeFileHandle fileHandle, Int64 offset, Int64 count, Int32 timeout, Int64& bytesSent) in /_/src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs:line 1905
   at System.Net.Sockets.SocketPal.SendFile(SafeSocketHandle handle, FileStream fileStream) in /_/src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs:line 933
   at System.Net.Sockets.Socket.SendFileInternal(String fileName, Byte[] preBuffer, Byte[] postBuffer, TransmitFileOptions flags) in /_/src/System.Net.Sockets/src/System/Net/Sockets/Socket.Unix.cs:line 148
   at System.Net.Sockets.Socket.SendFile(String fileName, Byte[] preBuffer, Byte[] postBuffer, TransmitFileOptions flags) in /_/src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs:line 1292
   at System.Net.Sockets.Socket.SendFile(String fileName) in /_/src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs:line 1271
   at System.Net.Sockets.Tests.SendFileTest.<>c__DisplayClass7_0.<SyncSendFileGetsCanceledByDispose>b__0() in /_/src/System.Net.Sockets/tests/FunctionalTests/SendFile.cs:line 209
   at System.Threading.Tasks.Task.InnerInvoke() in /_/src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs:line 2449
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj) in /_/src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs:line 2428
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs:line 172
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread) in /_/src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs:line 2385
   at System.Threading.Tasks.Task.ExecuteEntryUnsafe(Thread threadPoolThread) in /_/src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs:line 2327
   at System.Threading.Tasks.ThreadPoolTaskScheduler.<>c.<.cctor>b__10_0(Object s) in /_/src/System.Private.CoreLib/shared/System/Threading/Tasks/ThreadPoolTaskScheduler.cs:line 37
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state) in /_/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs:line 50
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs:line 172
   at System.Threading.ThreadHelper.ThreadStart(Object obj) in /_/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs:line 83
```